app/api.py

```diff
@@ -203,3 +204,24 @@ async def all_exception_handler(request: Request, exc: Exception):
             "message": "server error",
         },
     )
+
+
+@API.post("/financial_aid/{profile_id}")
+async def financial_aid(profile_id: str):
+    """Returns the the probability that financial aid will be required.
+
+    Calls the financial aid function from functions.py inputing the
+    profile_id for calculation involving formally incarcerated, low income,
+    and experience level as variables to formulate probability of financial aid
+
+    Args:
+        profile_id (str): the profile id of the mentee
+
+   Returns:
+        the probability that financial aid will be required
+    """
+
+    if profile_id not in API.db.read('Mentees', {"profile_id": profile_id}):
```

Comment:
This will always throw a 404 error as it can't find the profile_id in the API call. Everything else about your function seems to be working though.

Reply:
Fixed this error. Tested using my own mock db and data.